kolibri/auth/filters.py

```diff
@@ -0,0 +1,185 @@
+from django.db import models
+from django.db.models.query import F
+from six import string_types
+
+from .constants import collection_kinds
+
+
+class HierarchyRelationsFilter(object):
```

Comment:
> I think the implication is that we might create something that is not transferable to other database backends outside of sqlite -- is that true?

There's no flavor-specific SQL in there (i.e. not tied to sqlite) -- just standard joins and WHERE conditions. We do want to use Kolibri with other DB backends (like Postgres), but I don't think this code in particular is more vulnerable to DB differences than standard Django code (e.g. we've seen issues in KA Lite of different DB level integrity enforcement (such as uniqueness) in sqlite vs mysql. So, I think we'll want to (occasionally) run the full test suite using these alternative database engines, once we start using them, but don't think it makes sense to try to integrate them into the test suite itself (would probably make for much slower tests).

> I think this class in particular will benefit from comprehensive test coverage ... Can you get 100% coverage in #51? Otherwise I would feel obliged to spend more time on this PR.

Agreed. I will aim to get 100% coverage in #51, and delete anything that I'm not able to touch there.